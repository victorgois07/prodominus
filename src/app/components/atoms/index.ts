export * from "./Button";
export * from "./Error";
export * from "./Input";
export * from "./Label";
export * from "./Loading";
export * from "./Toast";
