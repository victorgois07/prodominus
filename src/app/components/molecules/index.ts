export * from "./CategoryFilter";
export * from "./ConfirmDialog";
export * from "./FormField";
export * from "./Modal";
export * from "./Pagination";
export * from "./SortFilter";
