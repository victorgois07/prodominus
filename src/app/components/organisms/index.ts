export * from "./FavoritesManagement";
export * from "./ProductCard";
export * from "./ProductFilters";
export * from "./ProductForm";
export * from "./ProductGrid";
export * from "./ProductManagement";
