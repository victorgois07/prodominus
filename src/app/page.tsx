"use client";

import { useProducts } from "@/business/hooks/useProducts";
import { useProductFilters } from "@/business/store/useProductFilters";
import {
  ConfirmDialog,
  Error,
  Loading,
  Pagination,
  ProductFilters,
  ProductManagement,
} from "./components";

export default function Home() {
  const { loading, error, totalPages } = useProducts();
  const { page, setPage } = useProductFilters();

  if (loading) return <Loading />;
  if (error) return <Error message={error} />;

  return (
    <main className="container mx-auto px-4 py-8">
      <h1 className="text-3xl font-bold text-gray-900 mb-8">Products</h1>

      <ProductFilters />

      <div className="mb-6 flex flex-col gap-4">
        <ProductManagement />
      </div>

      <Pagination page={page} totalPages={totalPages} onPage={setPage} />
      <ConfirmDialog />
    </main>
  );
}
