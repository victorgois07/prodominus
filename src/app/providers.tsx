"use client";

import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { useState } from "react";

export function Providers({ children }: { children: React.ReactNode }) {
  const [queryClient] = useState(
    () =>
      new QueryClient({
        defaultOptions: {
          queries: {
            staleTime: 1000 * 60 * 5,
            retry: 1,
            refetchOnWindowFocus: false,
            retryOnMount: true,
            refetchOnReconnect: true,
          },
        },
      })
  );

  return (
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  );
}
