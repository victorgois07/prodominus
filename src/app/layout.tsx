import type { Metadata } from "next";
import { Inter } from "next/font/google";
import { LoadingBackdrop } from "./components/atoms/LoadingBackdrop";
import "./globals.css";
import { Providers } from "./providers";

const inter = Inter({ subsets: ["latin"] });

export const metadata: Metadata = {
  title: "Prodominus - Product Management",
  description: "A modern product management system built with Next.js",
};

export default function RootLayout({
  children,
}: Readonly<{ children: React.ReactNode }>) {
  return (
    <html lang="en">
      <body className={`${inter.className} antialiased`}>
        <Providers>
          <header className="bg-white shadow-sm">
            <div className="container mx-auto px-4 py-4">
              <h1 className="text-2xl font-bold text-balance text-gray-900">
                Prodominus
              </h1>
            </div>
          </header>

          <main>{children}</main>

          <footer className="bg-gray-50 mt-8">
            <div className="container mx-auto px-4 py-6">
              <p className="text-center text-gray-600 text-sm">
                Â© {new Date().getFullYear()} Prodominus. All rights reserved.
              </p>
            </div>
          </footer>

          <LoadingBackdrop />
        </Providers>
      </body>
    </html>
  );
}
