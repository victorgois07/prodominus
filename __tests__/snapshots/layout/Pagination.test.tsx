import { Pagination } from "@/app/components";
import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

describe("Pagination Snapshot", () => {
  it("should match snapshot with first page", () => {
    const { container } = render(
      <Pagination page={1} totalPages={5} onPage={() => {}} />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with middle page", () => {
    const { container } = render(
      <Pagination page={3} totalPages={5} onPage={() => {}} />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with last page", () => {
    const { container } = render(
      <Pagination page={5} totalPages={5} onPage={() => {}} />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with many pages", () => {
    const { container } = render(
      <Pagination page={10} totalPages={20} onPage={() => {}} />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with single page", () => {
    const { container } = render(
      <Pagination page={1} totalPages={1} onPage={() => {}} />
    );

    expect(container).toMatchSnapshot();
  });
});
