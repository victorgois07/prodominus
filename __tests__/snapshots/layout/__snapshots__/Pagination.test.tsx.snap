// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pagination Snapshot > should match snapshot with first page 1`] = `
<div>
  <nav
    class="flex justify-center my-8"
  >
    <ul
      class="inline-flex items-center gap-1"
    >
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50"
          disabled=""
        >
          Anterior
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-green-500 text-white border border-green-600"
          disabled=""
        >
          1
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          2
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          3
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          4
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          5
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50"
        >
          Pr贸xima
        </button>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Pagination Snapshot > should match snapshot with last page 1`] = `
<div>
  <nav
    class="flex justify-center my-8"
  >
    <ul
      class="inline-flex items-center gap-1"
    >
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50"
        >
          Anterior
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          1
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          2
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          3
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          4
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-green-500 text-white border border-green-600"
          disabled=""
        >
          5
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50"
          disabled=""
        >
          Pr贸xima
        </button>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Pagination Snapshot > should match snapshot with many pages 1`] = `
<div>
  <nav
    class="flex justify-center my-8"
  >
    <ul
      class="inline-flex items-center gap-1"
    >
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50"
        >
          Anterior
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          1
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          2
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          3
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          4
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          5
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          6
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          7
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          8
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          9
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-green-500 text-white border border-green-600"
          disabled=""
        >
          10
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          11
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          12
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          13
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          14
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          15
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          16
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          17
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          18
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          19
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          20
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50"
        >
          Pr贸xima
        </button>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Pagination Snapshot > should match snapshot with middle page 1`] = `
<div>
  <nav
    class="flex justify-center my-8"
  >
    <ul
      class="inline-flex items-center gap-1"
    >
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50"
        >
          Anterior
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          1
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          2
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-green-500 text-white border border-green-600"
          disabled=""
        >
          3
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          4
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50 bg-white border border-gray-300"
        >
          5
        </button>
      </li>
      <li>
        <button
          class="px-4 py-2 rounded font-semibold transition disabled:opacity-50"
        >
          Pr贸xima
        </button>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Pagination Snapshot > should match snapshot with single page 1`] = `<div />`;
